tripy/tests/backend/test_compiler_api.py

```diff
@@ -95,7 +101,7 @@ def test_invalid_shape(self, shape, expected_error):
 @pytest.fixture(scope="session")
 def single_return_executable():
     compiler = tp.Compiler(add)
-    return compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32), tp.InputInfo((2, 2), dtype=tp.float32))
+    return compiler.compile(tp.InputInfo(tp.Shape([2, 2]), dtype=tp.float32), tp.InputInfo((2, 2), dtype=tp.float32))
```

Comment:
The idea was to support shape tensor inputs at runtime. I don't think we need to change anything about the compiler. `InputInfo` can be used to express the range of values the shape tensor can take.